proposals/storage-and-retrieval-bots.md

```diff
@@ -0,0 +1,168 @@
+			"Recv: 0 B, Paid 0 FIL, ClientEventProviderCancelled (DealStatusCancelling)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventDataTransferError (DealStatusErrored)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventOpen (DealStatusNew)".
+		],
+	"errorMessage":"ERROR: retrieval failed: Retrieve: Retrieval Error: error generated by data transfer: unable to send cancel to channel FSM: normal shutdown of state machine",
+	"tailLog":"....",            // Multiline, from `tail` of `lotus daemon`
+	"storageDealParameters":      // Given to RetrievalBot as input 
+		{
+			"CID":"Qm...",
+			"sha256":"73cb385...",    // independent checksum of data file
+			"sizeInBytes":"12345678".
+			"minderId":"f01924",
+			"verified":true,
+			"fastRetrievalFlag":true,
+			"dealId":"...",
+		}
+		"lotusVersion":"1.5.3-rc2+mainnet+git.9afb5ff94",
+												               // Call API `Filecoin.Version`
+		"datetime":"YYYY-MM-DD_HH:MM:SS",  // when attempt started
+},
+{
+  // ...next deal attempt json blob...
+}
+```
+
+Stdout will contain the results, in json, of each deal attempt.  It is intended to be piped into a log search service like those provided by AWS/GC.
+
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+ - Improve reliability of the network
+ - Enable an ecosystem of miner reputation and ranking systems
+ - Perform the retrieval verification in Slingshot 2.3
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+**Immensely!**
+
+ - We don't currently have enough information about why deals fail to allocate our debugging time and resources correctly.
+
+ - Miner reputation systems enabled by this tool would compliment the protocol-level incentives for miners to "do the right thing" (provide reliable retrieval of previously stored data, successfully complete all storage deals, etc)
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+C = 8
+
+Nothing is certain, but it is very likely that building this tool will at a minimum enable the Filecoin Project to better understand the frequency and causes of deal failures.  
+
+And the ability of this tool to support miner reputation systems can only help increase deals that get routed to reliable miners.
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+ - **Phase 1:  Retrieval Bot.**  Reads stdin describing a CID to attempt to retrieve, writes outcome of retrieval attempt to stdout.
+ - **Phase 2:  Storage Bot.**  Same idea but for storage deals.
+ - **Phase 3:  Orchestrator.**  Invokes the r-bot and s-bot programs with inputs one-by-one from a long queue of CIDs to test retrieve, or files to test store, etc.
```

Comment:
this probably is worth deferring to a second project of work from a scheduling perspective.

Reply:
Fixed.  I've linked to a draft PR for orchestrators to keep that separate.